Samples/WindowPlacement/README.md

```diff
@@ -0,0 +1,858 @@
+restarts, the app should relaunch the window Minimized, 'staying Minimized'
+over the app restart.
+
+Similarly, if the system reboots while the app is open and minimized, the window
+should be Minimized after reboot. (In other words, if a machine reboots
+overnight, all Minimized windows should not be restored from Minimize).
+
+As a general rule: when storing positions, remember the Minimize state. When
+launching normally, this state should be ignored (the window should launch to
+the normal position, or Maximized if WPF_RESTORETOMAXIMIZED). But when
+*relaunching* after a reboot or crash, relaunch to Minimized if it was
+previously Minimized.
+
+>![NOTE]
+>
+> A previous section described `STARTUPINFO`, which allows users to launch an
+appwith an explicit show command, e.g. `SW_MINIMIZE`. If launched in this way,
+like `start /min`, the app should launch Minimized regardless of the last close
+position.
+
+### Virtual Desktops
+
+Win+Tab and Taskview button on the taskbar allow the user to create
+multiple [**Virtual Desktops**](https://support.microsoft.com/en-us/windows/configure-multiple-desktops-in-windows-36f52e38-5b4a-557b-2ff9-e1a60c976434) (sometimes called **Desktops**). These are groups
+of windows that the user can switch between. When a window is on a background
+virtual desktop, it is cloaked (hidden).
+
+Apps generally do not need to worry about virtual desktops, including for most
+scenarios related to restoring window positions. In a normal app launch, it is
+always the case that the window should launch on the current virtual desktop
+(even if closed while on a background desktop).
+
+But, when restarted (automatic app/sytstem update, etc), apps should ideally
+'keep' their windows on the same virtual desktops. Consider an app like a
+browser that creates many windows, and a user has organized them onto different
+virtual desktops. If the system reboots and the app is restarted, it should
+restore all window positions, *and* it should move each window to its previous
+virtual desktop.
+
+This is done using the [`IVirtualDesktopManager`](https://learn.microsoft.com/en-us/windows/win32/api/shobjidl_core/nn-shobjidl_core-ivirtualdesktopmanager) APIs, which has these two functions:
+ - [`GetWindowDesktopId`](https://learn.microsoft.com/en-us/windows/win32/api/shobjidl_core/nf-shobjidl_core-ivirtualdesktopmanager-getwindowdesktopid)
+ - [`MoveWindowToDesktop`](https://learn.microsoft.com/en-us/windows/win32/api/shobjidl_core/nf-shobjidl_core-ivirtualdesktopmanager-movewindowtodesktop)
+
+When storing previous window positions, apps should remember the GUID for the
+window's virtual desktop. This is used ONLY when the app is restarted, not in a
+'normal launch'. This is similar to minimized; if closed while minimized and
+launched normally, the app should start restored (or maximized).
+
+### Restarting after Windows Update with `RegisterApplicationRestart`
```

Comment:
Relevant links:

* https://learn.microsoft.com/en-us/windows/win32/recovery/registering-for-application-restart 
* https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-registerapplicationrestart
* 
Ben Stolovitz
Dec 4




https://learn.microsoft.com/en-us/windows/apps/windows-app-sdk/applifecycle/applifecycle-restart 
* FYI https://github.com/nasbench/Misc-Research/blob/main/Other/Persistence-Via-RegisterAppRestart-Shim.md